catalog/plugin/catalog-plugin-metacardbackup-filestorage/src/main/java/org/codice/ddf/catalog/plugin/metacard/backup/storage/filestorage/MetacardFileStorageRoute.java

```diff
@@ -76,17 +81,25 @@ public void configure() throws Exception {
             METACARD_BACKUP_KEEP_DELETED_RTE_PROP,
             simple(String.valueOf(keepDeletedMetacards), Boolean.class))
         .choice()
-        .when(not(getCheckDeletePredicate()))
-        .stop()
-        .otherwise()
-        .choice()
         .when(not(getShouldBackupPredicate()))
         .stop()
         .otherwise()
         .setHeader(
             TEMPLATED_STRING_HEADER_RTE_PROP,
             method(new MetacardTemplate(outputPathTemplate), "applyTemplate(${body})"))
         .to("catalog:metacardtransformer")
+        .choice()
+        .when(
+            and(
+                header(PostIngestConsumer.ACTION).isEqualTo(PostIngestConsumer.DELETE),
+                getCheckDeletePredicate()))
+        .bean(
+            MetacardFileStorageRoute.class,
+            String.format(
+                "deleteFile(%s, ${in.headers.%s})",
```

Comment:
❗️ the starting dir and and filename should be URL encoded

Reply:
added encoding